R/DMMParam.R

```diff
@@ -0,0 +1,185 @@
+#' DMM clustering
+#' z
+#' Run the \code{DMM} function on a distance matrix within \code{\link{clusterRows}}.
+#' 
+#' @param k An integer indicating the number of clusters, or a list of integers defining the possible number of clusters to choose from.
+#' @param type A string specifying the fit function to use to find the optimal number of clusters. Use in combination with k being a list of integers
+#' @param type A boolean specifying 
+#' @param seed An integer specifying the seed to use when doing the DMM algorithm. Will be random if not set.
+#' 
+#' @inheritParams clusterRows
+#' @param BLUSPARAM A \linkS4class{KmeansParam} object.
+#' @param full Logical scalar indicating whether the clustering statistics from both steps should be returned.
+#' 
+#' @author Basil Courbayre
+#'
+#' @details
+#' To modify an existing DMMParam object \code{x},
+#' users can simply call \code{x[[i]]} or \code{x[[i]] <- value} where \code{i} is any argument used in the constructor.
+#'
+#' If \code{k=NULL}, the cluster search range will be 1:3.
+#' The default for the fit function is \code{laplace}.
+#' 
+#' @return 
+#' The \code{DMMParam} constructor will return a \linkS4class{DMMParam} object with the specified parameters.
+#'
+#' The \code{clusterRows} method will return a factor of length equal to \code{nrow(x)} containing the cluster assignments.
+#' If \code{full=TRUE}, a list is returned with \code{clusters} (the factor, as above) and \code{objects} 
+#' (a list containing \code{dmm}, the dmm object; \code{k}, the number of clusters; 
+#' \code{prob}, the array of probabilities; and \code{seed} the seed for the dmm).
+#'
+#' @examples
+#' clusterRows(iris[,1:4], DMMParam())
+#' clusterRows(iris[,1:4], DMMParam(k=2))
+#' clusterRows(iris[,1:4], DMMParam(k=1:3, type="laplace"))
+#'
+#' @name DMMParam-class
+#' @docType class
+#' @aliases 
+#' .defaultScalarArguments,DMMParam-method
+#' show,DMMParam-method
+NULL
+
+#' @export
+setClass("DMMParam", contains="BlusterParam", slots=c(k="integer_OR_NULL", 
+                                                      type="character",
+                                                      transposed="logical",
+                                                      seed = "integer_OR_NULL"))
+
+#' @export
+#' @rdname DMMParam-class
+DMMParam <- function(k=NULL, type=NULL, transposed=FALSE, seed=NULL) {
+    # Filling in missing values with the defaults.
+    current <- list(k=k, type=type, seed=seed)
+    notpresent <- vapply(current, is.null, FALSE)
+    if (any(notpresent)) {
+        defaults <- .get_dmm_defaults()
```

Comment:
This is essentially just calling a list --> remove function and use the list directly to reduce complexity